### PR TITLE
add ctests for 4DEnVAR and hybrid-4DEnVAR

### LINKED ISSUE #929: Add ctest for a hybrid 4DEnVAR to SOCA
## Description
Self-explanatory. Add a ctest to exercise a 4DEnVAR in SOCA with hybrid covariance.

Note that there is supposedly a bug with the hybrid 4D covariance. So answers will change at some point when that is fixed.

### PR DESCRIPTION
**Pull request by travissluka**
## Description

* closes https://github.com/JCSDA-internal/soca/issues/929

Adds 4d envar ctests
* 3 tests are added to perform short forecasts from the initial ensemble members so that I have the 4D state ensemble. (I didn't want to add more .nc files to the git-lfs)
* 4denvar test added that is pure ensemble 
* 4dhybenvar test that is a hybrid covariance.

I realize the 4denvar stuff is being worked on in oops, so I'm sure there will be subsequent changes to yaml and/or reference answers. Obviously, no one has looked at the science of this, yet.

### CODE REVIEW

File: test/testinput/4denvar.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
+cost function:
+  cost type: 4D-Ens-Var
+  analysis variables: [socn, tocn, uocn, vocn, ssh]
+  window begin: 2018-04-15T00:00:00Z
+  window length: PT6H
+  subwindow: PT6H  # NOTE, number of PEs has to be a multiple of num of windows!
```
Comment (kbhargava) [inline test/testinput/4denvar.yml:6]: Just a comment I wanted to put here: so far we can run soca tests with `--ntasak=1` but after this test we need `--ntasks=4`.

**Review by kbhargava — APPROVED** [review]
Looks good to me. I just quickly reviewed the code. Build and ran ctest on orion and they pass. In order to run the ctests for soca,  need to ask for more `ntasks=4` now. We might need to update the documentation accordingly.

**Review by guillaumevernieres — APPROVED** [review]
:+1:

**Comment by travissluka** [comment]
> Looks good to me. I just quickly reviewed the code. Build and ran ctest on orion and they pass. In order to run the ctests for soca, need to ask for more `ntasks=4` now. We might need to update the documentation accordingly.

we have documentation?!

**Pull request merged**